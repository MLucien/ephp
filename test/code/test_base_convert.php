<?php

class A {}

print "base_convert(101, 2, 10) = " . base_convert(101, 2, 10) . "\n";
print "base_convert(101, 2, '') = " . base_convert(101, 2, '') . "\n";
print "base_convert(101, '', 10) = " . base_convert(101, '', 10) . "\n";
print "base_convert(101, 1, 1) = " . base_convert(101, 1, 1) . "\n";
print "base_convert(101, 2, 1) = " . base_convert(101, 2, 1) . "\n";
print "base_convert('13031', 2, 10) = " . base_convert('13031', 2, 10) . "\n";
print "base_convert('AB10', 16, 10) = " . base_convert('AB10', 16, 10) . "\n";
print "base_convert(array(), 2, 2) = " . base_convert(array(), 2, 2) . "\n";
print "base_convert(new A, 2, 2) = " . base_convert(new A, 2, 2) . "\n";
print "end\n";
