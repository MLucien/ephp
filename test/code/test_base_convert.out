base_convert(101, 2, 10) = 5

Warning: base_convert() expects parameter 3 to be long, string given in {{CWD}}/test/code/test_base_convert.php on line 6
base_convert(101, 2, '') = 

Warning: base_convert() expects parameter 2 to be long, string given in {{CWD}}/test/code/test_base_convert.php on line 7
base_convert(101, '', 10) = 

Warning: base_convert(): Invalid `from base' (1) in {{CWD}}/test/code/test_base_convert.php on line 8
base_convert(101, 1, 1) = 

Warning: base_convert(): Invalid `to base' (1) in {{CWD}}/test/code/test_base_convert.php on line 9
base_convert(101, 2, 1) = 
base_convert('13031', 2, 10) = 5
base_convert('AB10', 16, 10) = 43792

Notice: Array to string conversion in {{CWD}}/test/code/test_base_convert.php on line 12
base_convert(array(), 2, 2) = 0

Catchable fatal error: Object of class A could not be converted to string in {{CWD}}/test/code/test_base_convert.php on line 13
